### PR TITLE
Add kubetest2 aks

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
Add kubetest2 aks
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: kubetest2-aks/cluster-templates/autoscaling-multipool.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            }
+          ],
+          "servicePrincipalProfile": {
+              "clientId": "{AZURE_CLIENT_ID}",
+              "secret": "{AZURE_CLIENT_SECRET}"
+          },
+          "encodedCustomConfiguration": "{CUSTOM_CONFIG}"
+        }
+      }
+
```
Comment (monianshouhou) [inline kubetest2-aks/cluster-templates/autoscaling-multipool.json:47]: There exist some indentation problems like line 46.

File: kubetest2-aks/cluster-templates/autoscaling.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            }
+          ],
+          "servicePrincipalProfile": {
+              "clientId": "{AZURE_CLIENT_ID}",
+              "secret": "{AZURE_CLIENT_SECRET}"
+          },
+          "encodedCustomConfiguration": "{CUSTOM_CONFIG}"
+        }
+      }
+
```
Comment (monianshouhou) [inline kubetest2-aks/cluster-templates/autoscaling.json:35]: ditto

File: .pipelines/scripts/run-e2e.sh [hunk 3]
```diff
@@ -62,26 +53,31 @@ elif [[ "${CLUSTER_TYPE}" == "autoscaling-multipool" ]]; then
...
-    "https://management.azure.com${AKS_CLUSTER_ID}?api-version=2022-04-02-preview" \
-    -d "$(cat ${CLUSTER_CONFIG_PATH})"
+export KUBECONFIG="${REPO_ROOT}/_kubeconfig/${RESOURCE_GROUP}_${CLUSTER_NAME}.kubeconfig"
+if [[ ! -f "${KUBECONFIG}" ]]; then
+  echo "kubeconfig not exists"
+  exit 1
+fi
+
+for i in `seq 1 6`; do
+  kubectl get pod -A && break
```
Comment (monianshouhou) [inline .pipelines/scripts/run-e2e.sh:77]: What's the propose of these lines? Maybe they should be moved after line 81.

**Review by monianshouhou — CHANGES_REQUESTED** [review]

File: .pipelines/scripts/run-e2e.sh (hunk 3 above)
Comment (lzhecheng) [reply .pipelines/scripts/run-e2e.sh:77]: This block of code is used to ensure cluster can be accessed with kubeconfig.

Comment (monianshouhou) [reply .pipelines/scripts/run-e2e.sh:77]: I understand. That's ok.

File: kubetest2-aks/cluster-templates/autoscaling-multipool.json (hunk 1 above)
Comment (lzhecheng) [reply kubetest2-aks/cluster-templates/autoscaling-multipool.json]: done

File: kubetest2-aks/cluster-templates/autoscaling.json (hunk 2 above)
Comment (lzhecheng) [reply kubetest2-aks/cluster-templates/autoscaling.json]: done

File: .pipelines/cnm-image.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+steps:
+  - task: GoTool@0
+    inputs:
+      version: '1.18.5'
+  - bash: |
+      git show --stat
+      echo $REGISTRY_PASSWORD | docker login $REGISTRY_URL -u $REGISTRY_USERNAME --password-stdin
+      export IMAGE_REGISTRY=$REGISTRY_URL
+      IMAGE_NAME="azure-cloud-controller-manager"
```
Comment (feiskyer) [inline .pipelines/cnm-image.yml:9]: should the image name be 'azure-cloud-node-manager'?

File: .pipelines/scripts/run-e2e.sh [hunk 5]
```diff
@@ -62,26 +53,31 @@ elif [[ "${CLUSTER_TYPE}" == "autoscaling-multipool" ]]; then
...
-    -H "AKSHTTPCustomFeatures: Microsoft.ContainerService/EnableCloudControllerManager" \
-    "https://management.azure.com${AKS_CLUSTER_ID}?api-version=2022-04-02-preview" \
-    -d "$(cat ${CLUSTER_CONFIG_PATH})"
+export KUBECONFIG="${REPO_ROOT}/_kubeconfig/${RESOURCE_GROUP}_${CLUSTER_NAME}.kubeconfig"
+if [[ ! -f "${KUBECONFIG}" ]]; then
+  echo "kubeconfig not exists"
+  exit 1
+fi
+
+for i in `seq 1 6`; do
```
Comment (feiskyer) [inline .pipelines/scripts/run-e2e.sh:76]: could you add a comment why get pods for a few times here?

File: kubetest2-aks/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+This is the kubetest2-aks repo to provision and delete aks clusters.
+
+## Commands
+Make kubetest2-aks binary
+```
+make install-deployer
+```
+
```
Comment (feiskyer) [inline kubetest2-aks/README.md:8]: Could you also add a list of environment variables that should be set before the following command?

File: .pipelines/cnm-image.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      IMAGE_TAG="$(git rev-parse --short=7 HEAD)"
```
Comment (monianshouhou) [inline .pipelines/cnm-image.yml:10]: Maybe variables like IMAGE_NAME and IMAGE_TAG here can be removed?

File: .pipelines/cnm-image.yml (hunk 4 above)
Comment (lzhecheng) [reply .pipelines/cnm-image.yml]: This variable should be removed.

File: .pipelines/scripts/run-e2e.sh (hunk 5 above)
Comment (lzhecheng) [reply .pipelines/scripts/run-e2e.sh:77]: Updated.

File: kubetest2-aks/README.md (hunk 6 above)
Comment (lzhecheng) [reply kubetest2-aks/README.md:8]: Updated.

File: .pipelines/cnm-image.yml (hunk 7 above)
Comment (lzhecheng) [reply .pipelines/cnm-image.yml]: Yes. Removed.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**